### ISSUE TITLE
[mysql 5.5] search request with bool not works

### ISSUE THREAD

**Issue by napley** [bolt/core#2307]
The generate request to search on checkbox field with a bool value is different between mysql 5.7 and 5.5.

With Mysql 5.7 :

[▼
  "evenements"
  "%true%"
  "online_event"
  "published"
  "start"
]

With Mysql 5.5 :

[▼
  "evenements"
  "["%true%"]"
  "online_event"
  "published"
  "start"
]

Details
-------

| Question                      | Answer
|----------------- | ------
| Relevant Bolt Version | 4.1.10
| Install type                   |  Composer install
| BC Break                     | no
| PHP version                 | 7.4
| Web server                  | Nginx
| For UX/UI issues        | Firefox 84

Reproduction
------------

A setContent request with search on a checkbox :

{% setcontent events = 'evenements' where { online_event: '%false%' } limit 6 orderby '-start' %}

**Comment by napley** [comment]
I made this https://github.com/bolt/core/commit/c81d328d845270a675a9810427e9268e2d0eac0e but I think its not very clean

**Comment by I-Valchev** [comment]
Hi @napley ,

Can you try this:

```twig
{% setcontent events = 'evenements' where { online_event: false } limit 6 orderby '-start' %}
```

**Comment by napley** [comment]
Hi,

With `{% setcontent events = 'evenements' where { online_event: true } limit 6 orderby '-start' %}` it works

but with `{% setcontent events = 'evenements' where { online_event: false } limit 6 orderby '-start' %}` I have this error:

An exception has been thrown during the rendering of a template ("No matching value found for """).

**Comment by I-Valchev** [comment]
@napley right, I am also getting the same error. This is a bug indeed

**Comment by I-Valchev** [comment]
hi @napley , I found how this can be solved. To do so, you need to edit the `QueryParameterParser.php` by adding the following four lines right under line 91:

```php
        $this->addValueMatcher("()", [
            'value' => '$1',
            'operator' => 'eq',
        ]);
```

I will be creating a PR to fix this, in the meantime you can manually override this for your local project.

**Comment by I-Valchev** [comment]
@napley can you please confirm what your bolt version is? I can only reproduce this on the master branch (which is for 4.2) but you mention above it breaks on 4.1.10?

**Comment by napley** [comment]
@I-Valchev It works with this fix for the version 4.1.10 :+1:

**Comment by I-Valchev** [comment]
Big thank you for confirming 👍
